source/opt/aggressive_dead_code_elim_pass.cpp

```diff
@@ -90,20 +90,34 @@ bool AggressiveDCEPass::AllExtensionsSupported() const {
   return true;
 }
 
-void AggressiveDCEPass::KillInstIfTargetDead(ir::Instruction* inst) {
+bool AggressiveDCEPass::KillInstIfTargetDead(ir::Instruction* inst) {
   const uint32_t tId = inst->GetSingleWordInOperand(0);
   const ir::Instruction* tInst = def_use_mgr_->GetDef(tId);
-  if (dead_insts_.find(tInst) != dead_insts_.end())
+  if (dead_insts_.find(tInst) != dead_insts_.end()) {
     def_use_mgr_->KillInst(inst);
+    return true;
+  }
+  return false;
+}
+
+void AggressiveDCEPass::ProcessLoad(uint32_t varId) {
+  // Only process locals
```

Comment:
This is fine.  You only need to process locals because the loads of non-locals are already included in the live set earlier in the flow.